src/libasr/codegen/wasm_to_x64.cpp

```diff
@@ -0,0 +1,136 @@
+#include <fstream>
+#include <chrono>
+#include <iomanip>
+
+#include <libasr/assert.h>
+#include <libasr/codegen/wasm_decoder.h>
+#include <libasr/codegen/wasm_to_x64.h>
+#include <libasr/codegen/x86_assembler.h>
+
+namespace LFortran {
+
+namespace wasm {
+
+/*
+
+This X64Visitor uses stack to pass arguments and return values from functions.
+Since in X64, instructions operate on registers (and not on stack),
+for every instruction we pop elements from top of stack and store them into
+registers. After operating on the registers, the result value is then
+pushed back onto the stack.
+
+One of the reasons to use stack to pass function arguments is that,
+it allows us to define and call functions with any number of parameters.
+As registers are limited in number, if we use them to pass function arugments,
+the number of arguments we could pass to a function would get limited by
+the number of registers available with the CPU.
+
+*/
+
+class X64Visitor : public WASMDecoder<X64Visitor>,
+                   public WASM_INSTS_VISITOR::BaseWASMVisitor<X64Visitor> {
+   public:
+    X86Assembler &m_a;
+
+    X64Visitor(X86Assembler &m_a, Allocator &al,
+               diag::Diagnostics &diagonostics, Vec<uint8_t> &code)
+        : WASMDecoder(al, diagonostics),
+          BaseWASMVisitor(code, 0U /* temporary offset */),
+          m_a(m_a) {
+        wasm_bytes.from_pointer_n(code.data(), code.size());
+    }
+
+    void gen_x64_bytes() {
+        // update the initial value of asm text as per X64 text format
+        m_a.update_asm("BITS 64\n\n");
+
+        emit_elf64_header(m_a);
+
+        {
+            m_a.add_label("_start");
+
+            // exit with a fixed non-zero exit code
+            m_a.asm_mov_r64_imm64(LFortran::X64Reg::rax, 60); // sys_exit
+            m_a.asm_mov_r64_imm64(LFortran::X64Reg::rdi, 24 /* exit_code */); // exit code
+            m_a.asm_syscall(); // syscall
+        }
+
+        emit_elf64_footer(m_a);
+    }
```

Comment:
Currently, we are generating fixed/pre-determined instructions.

The fixed instructions we generate in assembly textual format are
```asm
; setting exit code
mov rax, 60 ; sys_exit() system call
mov rdi, 24 ; exit code
syscall
```

Output:
```bash
(lp) lpython$ lpython examples/expr2.py --backend wasm_x64 -o tmp
(lp) lpython$ ./tmp
(lp) lpython$ echo $?
24
```

I tested with some other values  (>= `0` and <= `255` (`255` is probably max limit for exit code)) of exit codes. It seems to work.